airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -106,6 +106,10 @@ def read(
                         f"The requested stream {configured_stream.stream.name} was not found in the source."
                         f" Available streams: {stream_instances.keys()}"
                     )
+                stream_is_available, error = stream_instance.check_availability(logger, self)
+                if stream_is_available is False:
```

Comment:
nit for extra readability
```suggestion
                if not stream_is_available:
```
I find it safe as you documented check_availability returns `Tuple[bool, str]`. Feel free to ignore it if you think it's not safe :)

Reply:
Yes true! I had to change some of my pytest assertions to `is True` instead of `== True`, and then coming back to this part days later my brain must have mapped that onto here. But I could update those tests with this pattern too -- will do